recipe/build.sh

```diff
@@ -1,9 +1,15 @@
 #!/bin/bash
+set -ex
 
-set -x
+mkdir builddir
 
-# Use the G77 ABI wrapper everywhere so that the underlying blas implementation
-# can have a G77 ABI (currently only MKL)
-export SCIPY_USE_G77_ABI_WRAPPER=1
+# need to run meson first for cross-compilation case
+meson ${MESON_ARGS}
+    -Dblas=blas \
+    -Dlapack=lapack \
+    -Duse-g77-abi=true \
+    builddir || (cat builddir/meson-logs/meson-log.txt && exit 1)
```

Comment:
`builddir` is in the wrong place here. Can you remove it from this line, and make the first line:
```
meson setup builddir ${MESON_ARGS} \
```
(note that `\` was missing on line 7, which explains some failures.

Reply:
OK, I was going to come back to this PR today anyway 👍

Reply:
```
meson <args> builddir

meson setup <args> builddir

meson setup builddir <args>
```

Currently, this uses the first line, you're suggesting that only the third line is "correct". This is technically not true -- only the second line is "correct". It is in the `meson setup --help` text.

It's correct that neglecting the word "setup" is wrong -- meson warns about it now.

The `-D` words are options/option-arguments to "setup". The "builddir" word is an operand. Naming conventions come from the [POSIX / ISO 9945 utility conventions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html). Some people are more familiar with:
- options/option-arguments as, variously, "parameters", "arguments",  or "flags"
-  operands as "arguments" or "parameters".

Python's argparse, which Meson uses, calls:
- options -> "flags", "options", or "optional arguments"
- option-arguments -> "options", or "optional arguments"
- operands -> "positional arguments"

The POSIX utility conventions section 2, Utility Syntax Guidelines, state that options should always precede operands. GNU-style argument parsing differs from this -- it supports argument permutation, and whenever it finds an operand it shifts it toward the end and returns options/option-arguments first. argparse supports that, but not by default -- and Meson doesn't use `parse_intermixed_args()`, so we do not get the GNU style. But argparse *does* permit specifying *all* operands before options/option-arguments.

So returning to the 3 example Meson invocations -- argparse fundamentally supports both lines two and three, and treats them identically. But, the --help text will describe the use of line two.

...

tl;dr what to do?

Honestly, do whatever you want to do. But if it's a matter of technical correctness, leave "builddir" at the end. :D

Reply:
Thanks for the insights. For reference, https://mesonbuild.com/Cross-compilation.html has a call with the options after `builddir/`, and one without `setup`, so it's perhaps not yet homogeneously used in the docs.

Reply:
Ideally `setup` would be homogeneously used in the docs, but I could have possibly missed some when doing mass rewrites. :) It is of course totally wrong, because Meson will actively warn you if you follow the docs...

If you would like to submit a PR to fix the documentation wherever you catch stale references to a missing `setup`, that would be great. Otherwise I'll take a look when I get a chance.

(Again, the order of options simply doesn't matter, but there's a pedant's approach to correctness that suggests one way over the other, and no one will ever accuse me of not enjoying an excuse to pontificate about POSIX.)

Reply:
Good point, I inferred the needed order of arguments from the error message, but that was caused by a backslash missing and not by the position of `builddir`.